Gems/ROS2/Code/Source/Manipulation/JointStatePublisher.cpp

```diff
@@ -53,7 +54,12 @@ namespace ROS2
         m_jointStatePublisher->publish(m_jointStateMsg);
     }
 
-    void JointStatePublisher::OnTick(float deltaTime)
+    void JointStatePublisher::Activate()
+    {
+        InstallPhysicalCallback();
+    }
+
+    void JointStatePublisher::OnPhysicsSimulationFinished([[maybe_unused]] AzPhysics::SceneHandle sceneHandle, float deltaTime)
```

Comment:
I do not like that. This code will be call 
` JointsManipulationRequestBus::EventResult(
  result, m_context.m_entityId, &JointsManipulationRequests::GetJointPosition, joint name);`
with a very large frequency (1 kHz). It is doing a number of string and map operations that can be expensive, especially with a large number of joints.  Currently, the manipulator structure cannot be changed during simulation, the less elegant but performant solution needs to be initialised here.